app/assets/javascripts/helpers/i18n.js

```diff
@@ -18,7 +18,7 @@ Diaspora.I18n = {
   },
 
   updateLocale: function(locale, data) {
-    locale.data = $.extend(locale.data, data);
+    locale.data = $.extend({}, locale.data, data);
```

Comment:
So the improvement is the one I checked in this test: https://github.com/diaspora/diaspora/pull/7184/files#diff-bc361aa440611f178379efdd1012b25cR63.